spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -334,6 +359,116 @@ contract RolloverVault is
         return redemptions;
     }
 
+    /// @notice Swaps the bond tranche slices from the user into the underlying collateral for a fee,
+    ///         given that the vault has the remaining slices for immature redemption.
+    /// @dev When a user has some tranche slices from a given bond, and the vault has the remaining slices,
+    ///      this method allows the user and the vault to "meld" their assets together to redeem the underlying collateral
+    ///      from the bond. The user's share of the underlying is returned back and the vault charges
+    ///      the user a fee for providing liquidity.
+    /// @param bond The bond whose tranches are to be swapped.
+    /// @param trancheAmtsIn A list of amounts of each bond tranche the user is to deposit.
+    ///                      When the user does not have a particular slice, its amount in the list is to be set to zero.
+    /// @return The amount of underlying tokens returned.
+    function meld(IBondController bond, uint256[] memory trancheAmtsIn)
+        external
+        nonReentrant
+        whenNotPaused
+        returns (uint256)
+    {
+        // get bond tranches
+        BondTranches memory bt = bond.getTranches();
+
+        // * We check that the bond's underlying collateral token matches the vault's underlying.
+        // * We validate that the bond has at least 2 tranches to perform the meld operation.
+        // * We validate that the bond hasn't reached maturity as matching only works through immature redemption.
+        if (bond.collateralToken() != address(underlying) || bt.tranches.length <= 1 || bond.isMature()) {
```

Comment:
External contracts which interact with bw tranche contracts use `isMature` to check for if the bond is mature rather than checking against the maturity date (we treat it as a black box and just interact with it using the public interface). So I'm leaning toward keeping it as is ..

Reply:
Ok, I can see that for consistency 👍